apps/angular/ngfor-enhancement/src/app/app.component.ts

```diff
@@ -1,25 +1,71 @@
-import { NgFor, NgIf } from '@angular/common';
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Directive,
+  Input,
+  OnInit,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
 
 interface Person {
   name: string;
 }
 
+class ContextType<T> {
+  public get $implicit() {
+    return this.ngFor;
+  }
+  public ngFor!: T;
+}
+
+@Directive({
+  selector: '[ngFor]',
+  standalone: true,
+})
+export class NgForDirective<T> implements OnInit {
+  @Input('ngForOf') list!: Array<T>;
+  @Input('ngForEmpty') emptyList!: TemplateRef<any>;
+
+  constructor(
+    private viewContainerRef: ViewContainerRef,
+    private template: TemplateRef<any>,
+  ) {}
+
+  ngOnInit(): void {
+    if (this.list.length > 0) {
+      this.list.forEach((item) => {
+        const context = new ContextType<T>();
+        context.ngFor = item;
+        this.viewContainerRef.createEmbeddedView(this.template, context);
+      });
```

Comment:
you should delete this word to the real ngFor directive, you just want to enhance his behavior, not rewrite it.